### DISCUSSION TITLE
ESC button priority

### DISCUSSION THREAD

**Discussion by Marsony** [mantinedev/mantine#6973] [Q&A]
Good day to everyone, thank you very much for the wonderful library! I have a question: is it normal that when opening multiple modal windows in succession, pressing ESC closes the modal window that was opened first, rather than the one currently on top?

Wouldn't it be more logical, when multiple modal windows are open, for pressing ESC to close the window on top, rather than the one that was opened first (i.e., to close the second window, not the first)?

**Comment by rtivital** [comment]
This issue will be addressed in the next minor release

**Reply by enaluz** [reply]
Hello @rtivital, I just upgrade to `7.14.3` and I'm still experiencing this issue. Is the supposed fix included in this version?

**Reply by rtivital** [reply]
You need to use Modal.Stack component for this feature.
